### ISSUE TITLE
Respect `resume_download` deprecation

### ISSUE THREAD

**Issue by ssslakter** [UKPLab/sentence-transformers#2766]
In `SentenceTransformer(model_name)` there is a call which results this warning 
```
FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`
```
There were multiple issues in HF libraries that change the default parameter from False to None

**Comment by tomaarsen** [comment]
Hello!

Sentence Transformers does not use `resume_download` anywhere - this (very annoying) warning is thrown by the downstream dependency of `transformers` right here: https://github.com/huggingface/transformers/blob/ab0f050b42d903f34d6eb97f3f8c0c07f0517ad2/src/transformers/configuration_utils.py#L651

`transformers` hasn't made a release since 4.41.2 yet, so the issue on their part is only fixed on the `main` branch:
```
pip install -U git+https://github.com/huggingface/transformers.git
```
Or we can wait until the next release.

Apologies for the inconvenience, but I can't fix it on the sentence transformers side.

- Tom Aarsen

**Comment by ssslakter** [comment]
I see, thank you for the clarification! Is there a still open issue on the transformers side?

**Comment by tomaarsen** [comment]
No, the issue has been fixed there: https://github.com/huggingface/transformers/blob/83259e406d01966dc60c32d44cf14bb83d3ba8f5/src/transformers/configuration_utils.py#L651

It just hasn't been released yet.

- Tom Aarsen

**Comment by yudelevi** [comment]
Looks like this fix was released as part of transformers 4.42.0
